packages/dd-trace/src/config.js

```diff
@@ -234,7 +242,7 @@ class Config {
       process.env.DD_TRACE_EXPERIMENTAL_B3_ENABLED,
       false
     )
-    const defaultPropagationStyle = ['tracecontext', 'datadog']
+    const defaultPropagationStyle = inGCPFunction ? ['datadog', 'tracecontext'] : ['tracecontext', 'datadog']
```

Comment:
with google cloud functions, requests can include both tracecontext & datadog headers. this is because google cloud functions will always inject a tracecontext, even if a function is directly invoked.

In the even that a function is directly invoked by another datadog instrumented service upstream, we want datadog headers to be extracted (not tracecontext), so try to extract datadog headers first.

Reply:
Pretty sure this will break user expectations as the intent is for the _default_ propagation mechanism to be tracecontext which _should_ mean a user without config changes can deploy two services and they will talk to each other with tracecontext. If one of them happens to be in GCP this would change because you've swapped the ordering and it will therefore both expect datadog headers coming _in_, but also use datadog headers going _out_. It will not match the expectations of the other service despite seemingly not having any different configuration.

Reply:
removed per our discussions